blockchain/ethereum.go

```diff
@@ -328,7 +328,7 @@ func (e *EthereumClient) TransactionOpts(from *EthereumWallet) (*bind.TransactOp
 func (e *EthereumClient) ProcessTransaction(tx *types.Transaction) error {
 	var txConfirmer HeaderEventSubscription
 	if e.GetNetworkConfig().MinimumConfirmations == 0 {
-		txConfirmer = &InstantConfirmations{}
+		txConfirmer = NewL2TxConfirmer(e, tx.Hash())
```

Comment:
How come we have `NewL2TxConfirmer` if `e.GetNetworkConfig().MinimumConfirmations == 0`, is all networks with zero confirmations required are L2?

Reply:
They all essentially behave as Optimistic L2s, with near-instant confirmations. We shouldn't assume instant confirmation, this gets us in trouble with race conditions between our tests and the chain. (e.g. Instant confirming deployment of an OCR contract that we then try to configure a few ms before it exists on chain)